StratisCore.UI/src/app/wallet/receive/receive.component.ts

```diff
@@ -51,20 +65,30 @@ export class ReceiveComponent implements OnInit {
 
   private getUnusedReceiveAddresses(): void {
     const walletInfo = new WalletInfo(this.globalService.getWalletName());
-    this.apiService.getUnusedReceiveAddress(walletInfo)
-      .toPromise().then(
-      response => {
-        this.address = response;
-        // TODO: fix this later to use the actual sidechain name instead of 'cirrus'
-        const networkName = this.globalService.getSidechainEnabled() ? 'cirrus' : 'stratis';
-        this.qrString = `${networkName}:${response}`;
-      }
-    );
+    this.walletService.getUnusedReceiveAddress(walletInfo)
+      .toPromise()
+      .then(
+        response => {
+            this.address = response;
+            this.setQrString(response);
+        }
+      );
   }
 
-  private getAddresses(): void {
-    const walletInfo = new WalletInfo(this.globalService.getWalletName());
-    this.apiService.getAllAddresses(walletInfo)
+  private getAccountAddress() {
+    this.address = this.currentAccountService.getAddress();
+    this.setQrString(this.address);
+  }
+
+  private setQrString(address: string) {
+    // TODO: fix this later to use the actual sidechain name instead of 'cirrus'
+    const networkName = this.globalService.getSidechainEnabled() ? 'cirrus' : 'stratis';
```

Comment:
I wonder if we should move this into globalService (i.e. getNetworkName or something like that), so that later on if we have another sidechain we can easily change it in one place